File: SpriteKitLib/src/main/java/br/com/insanitech/spritekit/logger/Logger.java
package br.com.insanitech.spritekit.logger;

import java.util.logging.Level;

/**
 * Created by anderson on 6/29/15.
 */
public class Logger {
    public static boolean DEBUG_MODE = true;

    public static void log(String key, String content) {
        if (DEBUG_MODE) {
            getLogger().log(Level.INFO, key + " - " + content);
        }
    }

    public static void log(String content) {
        if (DEBUG_MODE) {
            getLogger().log(Level.INFO, content);
        }
    }

    private static java.util.logging.Logger getLogger() {
        final Throwable t = new Throwable();
        final StackTraceElement methodCaller = t.getStackTrace()[1];
        final java.util.logging.Logger logger = java.util.logging.Logger.getLogger(methodCaller.getClassName());
        logger.setLevel(Level.ALL);
        return logger;
    }
}

File: SpriteKitLib/src/main/java/br/com/insanitech/spritekit/logger/Logger.kt
package br.com.insanitech.spritekit.logger

import java.util.logging.Level

/**
 * Created by anderson on 6/29/15.
 */
object Logger {
    var DEBUG_MODE = true

    fun log(key: String, content: String) {
        if (DEBUG_MODE) {
            logger.log(Level.INFO, key + " - " + content)
        }
    }

    fun log(content: String) {
        if (DEBUG_MODE) {
            logger.log(Level.INFO, content)
        }
    }

    private val logger: java.util.logging.Logger
        get() {
            val t = Throwable()
            val methodCaller = t.stackTrace[1]
            val logger = java.util.logging.Logger.getLogger(methodCaller.className)
            logger.level = Level.ALL
            return logger
        }
}
